src/main/java/fi/vrk/xroad/monitor/actor/ResultCollectorActor.java

```diff
@@ -0,0 +1,110 @@
+package fi.vrk.xroad.monitor.actor;
+
+import akka.actor.AbstractActor;
+import fi.vrk.xroad.monitor.parser.SecurityServerInfo;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Actor for collecting security server monitoring data processing results
+ */
+@Component
+@Scope("prototype")
+@Slf4j
+public class ResultCollectorActor extends AbstractActor {
+
+  private Map<SecurityServerInfo, MonitorDataResult> trackResults;
+
+  public ResultCollectorActor(List<SecurityServerInfo> expectedResults) {
+    trackResults = new HashMap<>();
+    for (SecurityServerInfo securityServerInfo: expectedResults) {
+      trackResults.put(securityServerInfo, null);
+    }
+  }
+
+  @Override
+  public AbstractActor.Receive createReceive() {
+    return receiveBuilder()
+        .match(MonitorDataResult.class, this::handleMonitorDataResult)
+        .build();
+  }
+
+  private void handleMonitorDataResult(MonitorDataResult result) {
+    trackResults.put(result.getSecurityServerInfo(), result);
+    if (result.isSuccess()) {
+      log.info("received success with data {}", result.toString());
+    } else {
+      log.error("received error with data {}", result.toString());
+    }
+  }
+
+  @Override
+  public void postStop() throws Exception {
+    super.postStop();
+    for(Map.Entry e: trackResults.entrySet()) {
+      if (e.getValue() == null) {
+        log.error("Result for {} has not been received", e.getKey());
+      }
+    }
+  }
+
+  /**
+   * Indicates if all expected results have been received
+   */
+  public boolean isDone() {
```

Comment:
Maybe change visibility to default and add comment about these methods existing only for testing purposes? We do not want to accidentally mislead anyone unfamiliar with Akka into calling actors' methods directly.